tests/erc20_bridge_multihop.go

```diff
@@ -0,0 +1,450 @@
+}
+
+func submitCreateBridgeToken(
+	ctx context.Context,
+	source utils.SubnetTestInfo,
+	destinationChainID ids.ID,
+	destinationBridgeAddress common.Address,
+	nativeToken common.Address,
+	messageFeeAsset common.Address,
+	messageFeeAmount *big.Int,
+	fundedAddress common.Address,
+	fundedKey *ecdsa.PrivateKey,
+	transctor *erc20bridge.ERC20Bridge,
+	teleporterMessenger *teleportermessenger.TeleporterMessenger,
+) (*types.Receipt, *big.Int) {
+	opts, err := bind.NewKeyedTransactorWithChainID(fundedKey, source.ChainIDInt)
+	Expect(err).Should(BeNil())
+
+	tx, err := transctor.SubmitCreateBridgeToken(
+		opts,
+		destinationChainID,
+		destinationBridgeAddress,
+		nativeToken,
+		messageFeeAsset,
+		messageFeeAmount,
+	)
+	Expect(err).Should(BeNil())
+
+	// Wait for the transaction to be mined
+	receipt, err := bind.WaitMined(ctx, source.ChainRPCClient, tx)
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	event, err := utils.GetEventFromLogs(receipt.Logs, teleporterMessenger.ParseSendCrossChainMessage)
+	Expect(err).Should(BeNil())
+	Expect(event.DestinationChainID[:]).Should(Equal(destinationChainID[:]))
+
+	log.Info("Successfully SubmitCreateBridgeToken",
+		"txHash", tx.Hash().Hex(),
+		"messageID", event.Message.MessageID)
+
+	return receipt, event.Message.MessageID
+}
+
+func bridgeToken(
+	ctx context.Context,
+	source utils.SubnetTestInfo,
+	destinationChainID ids.ID,
+	destinationBridgeAddress common.Address,
+	nativeToken common.Address,
+	recipient common.Address,
+	totalAmount *big.Int,
+	primaryFeeAmount *big.Int,
+	secondaryFeeAmount *big.Int,
+	fundedAddress common.Address,
+	fundedKey *ecdsa.PrivateKey,
+	transctor *erc20bridge.ERC20Bridge,
+	isNative bool,
+	nativeTokenChainID ids.ID,
+	teleporterMessenger *teleportermessenger.TeleporterMessenger,
+) (*types.Receipt, *big.Int) {
+	opts, err := bind.NewKeyedTransactorWithChainID(fundedKey, source.ChainIDInt)
+	Expect(err).Should(BeNil())
+
+	tx, err := transctor.BridgeTokens(
+		opts,
+		destinationChainID,
+		destinationBridgeAddress,
+		nativeToken,
+		recipient,
+		totalAmount,
+		primaryFeeAmount,
+		secondaryFeeAmount,
+	)
+	Expect(err).Should(BeNil())
+
+	// Wait for the transaction to be mined
+	receipt, err := bind.WaitMined(ctx, source.ChainRPCClient, tx)
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	event, err := utils.GetEventFromLogs(receipt.Logs, teleporterMessenger.ParseSendCrossChainMessage)
+	Expect(err).Should(BeNil())
+	if isNative {
+		Expect(event.DestinationChainID[:]).Should(Equal(destinationChainID[:]))
+	} else {
+		Expect(event.DestinationChainID[:]).Should(Equal(nativeTokenChainID[:]))
+	}
+
+	return receipt, event.Message.MessageID
+}
+
+func approveBridgeToken(
+	ctx context.Context,
+	source utils.SubnetTestInfo,
+	bridgeTokenAddress common.Address,
+	transtor *bridgetoken.BridgeToken,
```

Comment:
nit:typo same for other functions in this file

Reply:
fixed